src/core/mempool.rs

```diff
@@ -675,18 +521,91 @@ impl MemPoolDB {
         Ok(rows)
     }
 
-    /// Get the next timestamp after this one that occurs in this chain tip.
-    pub fn get_next_timestamp(
+    /// Get all transactions at a specific block
+    #[cfg(test)]
+    pub fn get_num_tx_at_block(
         conn: &DBConn,
         consensus_hash: &ConsensusHash,
         block_header_hash: &BlockHeaderHash,
+    ) -> Result<usize, db_error> {
+        let sql = "SELECT * FROM mempool WHERE consensus_hash = ?1 AND block_header_hash = ?2";
+        let args: &[&dyn ToSql] = &[consensus_hash, block_header_hash];
+        let rows = query_rows::<MemPoolTxInfo, _>(conn, &sql, args)?;
+        Ok(rows.len())
+    }
+
+    /// Get the next nonce/timestamp after this nonce that occurs in the height range.
+    pub fn get_next_nonce(
```

Comment:
Minor style nit -- because this returns both a nonce and timestamp, this method name is misleading.  Can you call it something like `get_next_nonce_and_timestamp` or something similar?